CMakeLists.txt

```diff
@@ -1,69 +1,58 @@
-cmake_minimum_required(VERSION 3.0.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.21.0)
+if(POLICY CMP0177)
+  cmake_policy(SET CMP0177 NEW)
+endif()
 
-project (winflexbison)
+project (winflexbison C)
 
 if(NOT MSVC)
-   message( WARNING "Visual Studio Build supported only" )
+   message(FATAL_ERROR "Visual Studio Build supported only" )
 endif()
 
 add_definitions(-D_CRT_SECURE_NO_WARNINGS)
-
-if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-  add_definitions(-D_DEBUG)
-endif()
-
 # next line needed for compile in C (nor CPP) mode (ucrt headers bug)
 add_definitions(-Dinline=__inline)
 # next line needed for VS2017 only
 add_definitions(-Drestrict=__restrict)
 
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /W3 /MD /Od /Zi /EHsc")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /W3 /GL /Od /Oi /Gy /Zi /EHsc")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W3 /EHsc")
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /GL /Oi /Gy")
```

Comment:
changed this from CXX to C, since all projects are C projects. Those settings had zero effect before.
Shall I remove those lines (behavior unchanged) or use the new flags (lot of warnings, most likely unused intended behavior).

Reply:
Removed them, since they were previously unused. Changing them to C introduced warnings in code, from the original bison/flex project.